src/racer/nameres.rs

```diff
@@ -2450,8 +2462,57 @@ pub(crate) fn get_index_output(selfm: &Match, session: &Session) -> Option<Ty> {
         .next()
         .and_then(|(_, item_ty)| match item_ty {
             Ty::PathSearch(paths) => {
-                get_assoc_type_from_header(&paths.path, selfm, &index_header, session)
+                get_assoc_type_from_header(&paths.path, context, impl_header, session)
             }
             _ => Some(item_ty),
         })
 }
+
+/// Checks if trait_impl is the impl TraitName<OtherType>
+fn has_impl_for_other_type(trait_impl: &ImplHeader, trait_name: &str, other_type: &str) -> bool {
+    if let Some(ref path) = trait_impl.trait_path() {
+        if path.name() == Some(trait_name) {
+            if path.segments[0].generics.len() > 0 {
+                return match path.segments[0].generics[0] {
+                    Ty::PathSearch(ref generic) => Some(other_type) == generic.path.name(),
+                    _ => false,
+                };
+            }
+            // default is self
+            return trait_impl.self_path().name() == Some(other_type);
+        }
+    }
+    false
+}
+
+/// Resolves the type of a binary expression
+/// # Arguments
+/// * base_type: the type on the left hand side
+/// * node: the operator
+/// * other_type: the type on the right hand side
+/// *Note*: Works only for std::ops::* now
+pub(crate) fn resolve_binary_expr_type(
+    base_type: &Match,
+    node: BinOpKind,
+    other_type: &str,
```

Comment:
This should be `Option<&str>`.
I think we should inference that `L + R = L` if we don't know R's type.

Reply:
Does it make sense to check if there's an override impl to reduce false positives?

Reply:
I don't think so.
The important point is **we can't always find operator impl**.
(e.g. If we use [derive_more](https://github.com/JelteF/derive_more) and write
```rust
#[derive(Add, Sub)]
struct S(u64);
```, or place impl in other file than struct definition file).
So I think to assume that L=R  is OK if
- we can't find R's type
- we can'T find trait implementation